staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/informers/externalversions/factory.go

```diff
@@ -167,10 +180,22 @@ func (f *sharedInformerFactory) InformerFor(obj runtime.Object, newFunc internal
 
 // SharedInformerFactory provides shared informers for resources in all known
 // API group versions.
+//
+// It is typically used like this:
+//  1. create a factory
+//  2. create one or more informers
+//  3. call Start (from the underlying interface) with a stop channel to create
+//     goroutines which handle the informers
+//  4. close the stop channel
+//  5. optional: WaitForShutdown to ensure that goroutines have stopped
```

Comment:
I think example code would be helpful than descriptions for this. Could you add an example instead